test/integration/logging_test.go

```diff
@@ -0,0 +1,236 @@
+	if tokenErr != nil {
+		fmt.Printf("could not create a token, %s", tokenErr)
+		t.Fatal(tokenErr)
+	}
+
+	clientHTTP := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	kac := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "KubeArchiveConfig",
+			"apiVersion": fmt.Sprintf("%s/%s", kubearchivev1alpha1.SchemeBuilder.GroupVersion.Group, kubearchivev1alpha1.SchemeBuilder.GroupVersion.Version),
+			"metadata": map[string]string{
+				"name":      "kubearchive",
+				"namespace": namespaceName,
+			},
+			"spec": map[string]any{
+				"resources": []map[string]any{
+					{
+						"selector": map[string]string{
+							"apiVersion": "v1",
+							"kind":       "Pod",
+						},
+						"archiveWhen": "true",
+						"deleteWhen":  "status.phase == 'Succeeded'",
+					},
+				},
+			},
+		},
+	}
+
+	gvr := kubearchivev1alpha1.GroupVersion.WithResource("kubearchiveconfigs")
+	_, kacErr := dynamicClient.Resource(gvr).Namespace(namespaceName).Create(context.Background(), kac, metav1.CreateOptions{})
+	if kacErr != nil {
+		t.Fatal(kacErr)
+	}
+
+	// Give a bit of time to KubeArchive to pickup filters changes
+	time.Sleep(5 * time.Second)
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fedora",
+			Namespace: namespaceName,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Containers: []corev1.Container{
+				corev1.Container{
+					Name:    "fedora",
+					Command: []string{"echo", "Hello World"},
+					Image:   "quay.io/fedora/fedora:latest",
+				},
+			},
+		},
+	}
+
+	_, createPodErr := clientset.CoreV1().Pods(namespaceName).Create(context.Background(), pod, metav1.CreateOptions{})
+	if createPodErr != nil {
+		t.Fatal(createPodErr)
+	}
+
+	url := fmt.Sprintf("https://localhost:8081/api/v1/namespaces/%s/pods", namespaceName)
+	var list *unstructured.UnstructuredList
+	var getUrlErr error
+	retryErr = retry.Do(func() error {
+		list, getUrlErr = getUrl(&clientHTTP, token.Status.Token, url)
+		if getUrlErr != nil {
+			t.Fatal(getUrlErr)
+		}
+
+		if len(list.Items) == 1 {
+			return nil
+		}
+
+		return fmt.Errorf("received '%d' elements, need just 1", len(list.Items))
+	}, retry.Attempts(10), retry.MaxDelay(4*time.Second))
+
+	if retryErr != nil {
+		t.Fatal(retryErr)
+	}
+
+	podName := list.Items[0].GetName()
+	url = fmt.Sprintf("https://localhost:8081/api/v1/namespaces/%s/pods/%s/log", namespaceName, podName)
+	request, errRequest := http.NewRequest("GET", url, nil)
+	if errRequest != nil {
+		fmt.Printf("could not create a request, %s", errRequest)
+		t.Fatal(errRequest)
+	}
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Status.Token))
+	fmt.Printf("HTTP request: %s\n", request.URL.RequestURI())
```

Comment:
I'd suggest moving the helper function of `getUrl()` in `integration_test.go` to `test/main.go` so it's available to every integration test (same as the PortForward function) and use it here too.

It will imply not doing the "parsing" of the data to an `Unstructured` struct in the function but I think the refactor is worth it.

Reply:
I'll see what I can do.

Reply:
I moved the function, but I didn't change the implementation. Most calls expect to parse the body into a unstructured list, so I leave it as is. The logging test has to implement its own thing to check the logs.